src/web/services/instructor-help-data-sharing.service.ts

```diff
@@ -0,0 +1,29 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable } from 'rxjs';
+
+/**
+ * Handles sharing data to manage elements to be collapsed when following anchor tags.
```

Comment:
Why we need service to handle collapse event?

Reply:
Yes, it also looks a bit like overkill to me. 

However, the instructor help page consists of several components and, as I mentioned in the PR description, I did not manage to find a better way to handle/trigger collapse event in component Y from component X. I am a beginner in Angular development, though, and I will be happy to try alternative approaches if any!

Reply:
Why not trying `componentA -(event)-> parentComponent -(update)-> componentB`

Reply:
Ah yes, makes sense!

Changes made, please see the last commits below.